app/jobs/concerns/maintenance_tasks/task_job_concern.rb

```diff
@@ -58,7 +69,8 @@ def build_enumerator(_run, cursor:)
     def each_iteration(input, _run)
       throw(:abort, :skip_complete_callbacks) if @run.stopping?
       task_iteration(input)
-      @ticker.tick
+      batch_size = input.is_a?(ActiveRecord::Relation) ? input.size : 1
```

Comment:
This requires an unnecessary `COUNT(*)` to show records instead of batches in the progress, since the batch is not `loaded?` when it comes out of JobIteration.
Maybe we could somehow re-use the batch size and have some rounding error in the number of ticks recorded, or we should use the number of batches directly (and I'm not sure if we should explain that in the UI, but I think we have currently of persisting the information that the Task returned a relation in batches).

Reply:
Yes we can, there would be some rounding error where if there's 7 records and you have batches of 5, you'd see 10 records processed at the end.

Reply:
Changes incoming! 😁